### ISSUE TITLE
transform it into a library to use instead of cmdline

### ISSUE THREAD

**Issue by jedi4ever** [thbishop/fission#5]
I've just created a fog provider for vmfusion via fission. 

See - https://github.com/geemus/fog/commit/1a576820ae34133e1cba2f56f8b606aec182208d for details

a lot of the parts of the library use the ui.output and the exit (instead of raise) . It might be useful to revise the strategy or at least be able to silence the ui.

**Comment by thbishop** [comment]
I've been toying with a couple of different ideas to refactor for usage as a lib.  Once I get 0.3.0 final pushed out, I'll start working on this type of support for 0.4.0.

**Comment by thbishop** [comment]
@jedi4ever I just pushed a branch with an implementation I've been playing with for this.  The basic idea is that all of the operations will return a response object.  The response object will have information about the operation: if it was successful, output (if any; typically just for errors), and data (string, boolean, array, etc.).  As the operations return a response object, they will no longer be responsible for display or decision on exiting (this is left for the caller to handle).

Here's a comparison of the changes: https://github.com/thbishop/fission/compare/master...use_as_lib

There's still a bunch of cleanup I need to do, but I think you can get the basic idea.

Any feedback would be greatly appreciated.

**Comment by jedi4ever** [comment]
Hey great stuff! 

I've merged my old changes to a new branch 'enhance_use_as_lib' in my fork.

Some ideas/remarks:
- for something as exists? it does make more sense to return a boolean and not a fusion response object
- for something like VM.all is makes more sense to return the names without the response object
- I would use the response object only for the actions such as stop,start,suspend ...
- on clone I have mixed feelings -> should it return a vm object? or a response.

Let me know what you think on this , I'm set to do some work on fission today :)
